src/ngraph/graph_util.cpp

```diff
@@ -165,6 +165,7 @@ void ngraph::replace_node(std::shared_ptr<Node> target, std::shared_ptr<Node> re
             input->replace_output(replacement->get_outputs().at(i));
         }
     }
+    replacement->merge_provenance_tags_from(target);
```

Comment:
Yeah, this thing here is exactly the piece that will get more complex as we go. :)